h2/src/main/org/h2/mvstore/MVStore.java

```diff
@@ -1735,7 +1744,9 @@ public int compare(Chunk o1, Chunk o2) {
                 }
             }
             if (!queue.isEmpty()) {
-                result = queue;
+                ArrayList<Chunk> list = new ArrayList<>(queue);
+                Collections.sort(list, Chunk.PositionComparator.INSTANCE);
```

Comment:
Here the sorting throws out the above sort of movePriority.

Reply:
This is intentional. Selection is done at this point, and queue's order was for selection purposes, and new order defines order of chunks movement - plain left-to-right.

Reply:
OK, here pass. But sort may be overhead.

Reply:
Overhead is expected to be minimal compare to the cost of actual chunk movement